guardrails/guard.py

```diff
@@ -1074,6 +1074,12 @@ def with_prompt_validation(
         Args:
             validators: The validators to add to the prompt.
         """
+        warnings.warn(
+            """The `with_prompt_validation` method is deprecated,
+            and will be removed in 0.5.x. Instead, please use
+            `Guard().use(YourValidator, on='prompt')`.""",
+            FutureWarning,
+        )
```

Comment:
Since this is a method the user is calling, let's decorate this with `typing_extension.deprecated` so it shows with a ~strikethrough~ in most IDE's when they use it.